panel/io/cache.py

```diff
@@ -305,11 +308,46 @@ def compute_hash(func, hash_funcs, args, kwargs):
         _HASH_MAP[key] = hash_value
     return hash_value
 
+P = ParamSpec("P")
+R = TypeVar("R")
+T = TypeVar("T")
+
+@overload
+def cache(
+    func: Literal[None] = ...,
+    hash_funcs: dict[type[Any], Callable[[Any], bytes]] | None = ...,
+    max_items: int | None = ...,
+    policy: Literal['FIFO', 'LRU', 'LFU'] = ...,
+    ttl: float | None = ...,
+    to_disk: bool = ...,
+    cache_path: str = ...,
+    per_session: bool = ...,
+) -> Callable[[Callable[P, R]], Callable[P, R]]:
+    ...
+
+@overload
+def cache(
+    func: Callable[P, R],
+    hash_funcs: dict[type[Any], Callable[[Any], bytes]] | None = ...,
+    max_items: int | None = ...,
+    policy: Literal['FIFO', 'LRU', 'LFU'] = ...,
+    ttl: float | None = ...,
+    to_disk: bool = ...,
+    cache_path: str = ...,
+    per_session: bool = ...,
+) -> Callable[P, R]:
+    ...
 
 def cache(
-    func=None, hash_funcs=None, max_items=None, policy='LRU',
-    ttl=None, to_disk=False, cache_path='./cache', per_session=False
-):
+    func: Callable[P, R] | None = None,
+    hash_funcs: dict[type[Any], Callable[[Any], bytes]] | None = None,
+    max_items: int | None = None,
+    policy: Literal['FIFO', 'LRU', 'LFU'] = 'LRU',
+    ttl: float | None = None,
+    to_disk: bool = False,
+    cache_path: str = './cache',
```

Comment:
As far as I can see `cache_path` is used as input to `os.path.join`. Thus I believe that it could also be a `pathlib.Path` type and that the type annotation should be updated to this?

Reply:
I _think_ `str | os.PathLike` is right.